selftest/selftest

```diff
@@ -4,22 +4,50 @@ set -e
 
 if [ -x "$(command -v fgconsole)" ]
 then
-    trap "sudo chvt $(sudo fgconsole)" EXIT
+    trap "sudo chvt $(sudo fgconsole); sudo snap connect mir-test-tools:x11" EXIT
 fi
 
+export MIR_SERVER_WAYLAND_HOST="${WAYLAND_DISPLAY:wayland-0}"
+unset WAYLAND_DISPLAY
+
 if [ -v DISPLAY ]
 then
+    export MIR_SERVER_PLATFORM_DISPLAY_LIBS=mir:x11
+    sudo snap connect mir-test-tools:x11
+
+    echo "**** INFO: running tests on mir:x11 ****"
     snap run mir-test-tools.gtk3-test
     snap run mir-test-tools.qt-test
     snap run mir-test-tools.sdl2-test
     snap run mir-test-tools.smoke-test
-    snap run mir-test-tools.performance-test
+    snap run mir-test-tools.performance-test --gtest_filter=-GLMark2Xwayland.*
 fi
 
+unset DISPLAY
+
+if [ "$XDG_SESSION_TYPE" = "wayland" ] && [ -O "${MIR_SERVER_WAYLAND_HOST}" ]
```

Comment:
Don't think `-O` is what you want here?

```
       -O FILE
              FILE exists and is owned by the effective user ID
```

Reply:
What do you think I want? (Maybe there's something I didn't think of)

Reply:
Well, unless it's a full path, it will never test true?

Usually we're doing `[ -n …]`?

Reply:
Depends what we're testing. C.f. https://github.com/canonical/mir-test-tools/blob/7e593d47308ed20608d634b8b2525bb1df12bc2a/wrapper/server-wrapper#L4

Reply:
So how do I ensure that `selftest` actually runs on wayland? Do I set `MIR_SERVER_WAYLAND_HOST` to a full path?

Reply:
> Well, unless it's a full path, it will never test true?

Fixed in latest commit. (I screwed up a clean-up edit before pushing)